assignments/lambda-classes.js

```diff
@@ -1 +1,110 @@
 // CODE here for your Lambda Classes
+
+class Person {
+  constructor(element) {
+    this.name = element.name;
+    this.age = element.age;
+    this.location = element.location;
+    this.gender = element.gender;
+  }
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(element) {
+    super(element);
+    this.specialty = element.specialty;
+    this.favLanguage = element.favLanguage;
+    this.catchPhrase = element.catchPhrase;
+  }
+
+  demo(subject) {
+    console.log(`Today we are learning about ${subject}`);
+  }
+
+  grade(student, subject) {
+    console.log(`${student.name} receives a perfect score on ${subject}`);
+  }
+}
+
+class Student extends Person {
+  constructor(element) {
+    super(element);
+    this.previousBackground = element.previousBackground;
+    this.className = element.className;
+    this.favSubjects = element.favSubjects;
+  }
+
+  listsSubjects() {
+    this.favSubjects.forEach(element => {
+      console.log(element);
+    });
+  }
+
+  PRAssignment(subject) {
```

Comment:
`student.name` is undfined because you're not passing anything. If you want the studen'ts name in the same constructor, you must add `this.name`

I refractored both methods:

```
PRAssignment(subject) {
    console.log(`${this.name} has submitted a PR for ${subject}`);
  }

  sprintChallenge(subject) {
    console.log(`${this.name} has begun sprint challenge on ${subject}`);
  }
```